algorithm.go

```diff
@@ -72,6 +75,44 @@ func (a Algorithm) String() string {
 	}
 }
 
+func (a Algorithm) IsSupported() bool {
```

Comment:
Missing documentation. Actually, I am a bit confused on how "supported" is defined.